mlflow/sklearn/utils.py

```diff
@@ -174,81 +203,169 @@ def _get_classifier_metrics(fitted_estimator, fit_args, fit_kwargs):
-            name=_METRICS_PREFIX + "precision_score",
+        _SklearnMetricOrArtifact(
+            name=_TRAINING_PREFIX + "precision_score",
             function=sklearn.metrics.precision_score,
             arguments=dict(
                 y_true=y_true, y_pred=y_pred, average="weighted", sample_weight=sample_weight
             ),
+            title="",
         ),
-        _SklearnMetric(
-            name=_METRICS_PREFIX + "recall_score",
+        _SklearnMetricOrArtifact(
+            name=_TRAINING_PREFIX + "recall_score",
             function=sklearn.metrics.recall_score,
             arguments=dict(
                 y_true=y_true, y_pred=y_pred, average="weighted", sample_weight=sample_weight
             ),
+            title="",
         ),
-        _SklearnMetric(
-            name=_METRICS_PREFIX + "f1_score",
+        _SklearnMetricOrArtifact(
+            name=_TRAINING_PREFIX + "f1_score",
             function=sklearn.metrics.f1_score,
             arguments=dict(
                 y_true=y_true, y_pred=y_pred, average="weighted", sample_weight=sample_weight
             ),
+            title="",
         ),
-        _SklearnMetric(
-            name=_METRICS_PREFIX + "accuracy_score",
+        _SklearnMetricOrArtifact(
+            name=_TRAINING_PREFIX + "accuracy_score",
             function=sklearn.metrics.accuracy_score,
             arguments=dict(
                 y_true=y_true, y_pred=y_pred, normalize=True, sample_weight=sample_weight
             ),
+            title="",
         ),
     ]
 
     if hasattr(fitted_estimator, "predict_proba"):
         y_pred_proba = fitted_estimator.predict_proba(X)
         classifier_metrics.extend(
             [
-                _SklearnMetric(
-                    name=_METRICS_PREFIX + "log_loss",
+                _SklearnMetricOrArtifact(
+                    name=_TRAINING_PREFIX + "log_loss",
                     function=sklearn.metrics.log_loss,
                     arguments=dict(y_true=y_true, y_pred=y_pred_proba, sample_weight=sample_weight),
+                    title="",
                 ),
             ]
         )
 
         if _is_metric_supported("roc_auc_score"):
+            # For binary case, the parameter `y_score` expect scores must be
+            # the scores of the class with the greater label.
+            if y_pred_proba.shape[1] == 2:
+                y_pred_proba_binary = [prob[1] for prob in y_pred_proba]
+
             classifier_metrics.extend(
                 [
-                    _SklearnMetric(
-                        name=_METRICS_PREFIX + "roc_auc_score",
+                    _SklearnMetricOrArtifact(
+                        name=_TRAINING_PREFIX + "roc_auc_score",
                         function=sklearn.metrics.roc_auc_score,
                         arguments=dict(
                             y_true=y_true,
-                            y_score=y_pred_proba,
+                            y_score=y_pred_proba_binary,
```

Comment:
This throws a "variable referenced before assignment" exception when the classifier is non-binary (because `y_pred_proba_binary` never gets assigned in that case). Also, can you explain the motivation for making this change? I tried this out with logistic regression on the Iris dataset and got the exact same value for `roc_auc_score` when using `y_pred_proba` and `y_pred_proba_binary`. I'd rather not make this change if we don't have to, especially because we're making a lot of assumptions about `y_pred_proba` (i.e. it has a `shape` attribute and has the shape `(?, 2)`).

Reply:
Oh I messed up with the logic.
But regarding the motivation, we can talk about this during sync-up.